### PR TITLE
Add BasemapGallery toolkit component sample. 

### PR DESCRIPTION
**Pull request by asantana917**

### CODE REVIEW

**Review by nCastle1 — CHANGES_REQUESTED** [review]
Sample looks good, I'm glad to see the BasemapGallery toolkit component being used here.

Feedback:

- metadata.json should reference the Toolkit package
- You don't need to use the BasemapSelected event. It would be better to bind the `BasemapGallery` to the `MapView`'s map via the `GeoModel` bindable property. (readme will need to be updated)
- The BasemapGallery does not require a Portal to be set, it will use ArcGIS Online by default. Once the binding is set up and a map is added to the MapView, there will be no need for the Initialize and BasemapSelected methods.

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Map/ChangeBasemap/readme.md [hunk 1]
```diff
@@ -10,17 +10,19 @@ Basemaps should be selected contextually. For example, in maritime applications,
...
 1. Create an `Map` object.
 2. Set the map to the `MapView` object.
-3. Choose a new basemap type and set it on the map.
+3. Create a `BasemapGallery` using the toolkit. 
+4. Use the `BasemapSelected` event to change the basemap. 
 
 ## Relevant API
 
 * Basemap
+* BasemapGallery
```
Comment (ZackAllen) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Map/ChangeBasemap/readme.md:25]: We should add tags for toolkit and gallery.

File: src/Forms/Shared/Samples/Map/ChangeBasemap/ChangeBasemap.xaml [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
-  <Grid>
-    <Grid.RowDefinitions>
-      <RowDefinition Height="*" />
-      <RowDefinition Height="auto" />
-    </Grid.RowDefinitions>
-    <esriUI:MapView x:Name="MyMapView" />
-    <Button x:Name="changeBasemapButton" Text="Change Basemap" Clicked="OnChangeBasemapButtonClicked" Grid.Row="1"/>
-  </Grid>
+             xmlns:toolkit="clr-namespace:Esri.ArcGISRuntime.Toolkit.Xamarin.Forms;assembly=Esri.ArcGISRuntime.Toolkit.Xamarin.Forms">
+    <Grid>
```
Comment (ZackAllen) [inline src/Forms/Shared/Samples/Map/ChangeBasemap/ChangeBasemap.xaml:8]: The behavior of the UI isn't super intuitive.
Maybe hide the basemap options after a selection is made?
Or change the button text when the gallery is displayed (i.e. flip between "Show basemaps" and "Hide basemaps".

File: src/Forms/Shared/Samples/Map/ChangeBasemap/ChangeBasemap.xaml [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
+        <Grid.RowDefinitions>
+            <RowDefinition Height="*" />
+            <RowDefinition Height="auto" />
+        </Grid.RowDefinitions>
+        <esriUI:MapView x:Name="MyMapView" />
+        <toolkit:BasemapGallery x:Name="MyBasemapGallery" BasemapSelected="BasemapSelected" />
```
Comment (ZackAllen) [inline src/Forms/Shared/Samples/Map/ChangeBasemap/ChangeBasemap.xaml:14]: The transparent background can make the text hard to read. I would add an opaque background here.

File: src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Map/ChangeBasemap/ChangeBasemap.xaml.cs [hunk 4]
```diff
@@ -18,50 +19,31 @@ namespace ArcGISRuntime.WPF.Samples.ChangeBasemap
...
-            {"OpenStreetMap", new Basemap(BasemapStyle.OSMStandard)}
-        };
-
         public ChangeBasemap()
         {
             InitializeComponent();
 
-            // Create the UI, setup the control references and execute initialization
+            // Create the UI, setup the control references and execute initialization.
             Initialize();
```
Comment (ZackAllen) [inline src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Map/ChangeBasemap/ChangeBasemap.xaml.cs:31]:
```suggestion
            _ = Initialize();
```
Needed for the warning.

**Review by ZackAllen — CHANGES_REQUESTED** [review]

File: src/Forms/Shared/Samples/Map/ChangeBasemap/ChangeBasemap.xaml [hunk 5]
```diff
@@ -11,10 +11,13 @@
             <RowDefinition Height="auto" />
         </Grid.RowDefinitions>
         <esriUI:MapView x:Name="MyMapView" />
-        <toolkit:BasemapGallery x:Name="MyBasemapGallery" BasemapSelected="BasemapSelected" />
-        <Button x:Name="changeBasemapButton"
+        <toolkit:BasemapGallery x:Name="MyBasemapGallery"
+                                BackgroundColor="White"
+                                GeoModel="{Binding Source={x:Reference MyMapView}, Path=Map}"
+                                Opacity="0.85" />
```
Comment (nCastle1) [inline src/Forms/Shared/Samples/Map/ChangeBasemap/ChangeBasemap.xaml:17]: Opacity shouldn't be set this way. It ends up being inherited by subcomponents of the view, which leads to unexpected behavior with the templates. I think the UI is clear enough with the toggle button.

**Review by nCastle1 — CHANGES_REQUESTED** [review]
Looking good, just one last issue to fix.

**Review by nCastle1 — APPROVED** [review]

**Review by ZackAllen — APPROVED** [review]

**Pull request merged**